modules/mapbox/src/deck-utils.ts

```diff
@@ -158,13 +180,18 @@ function getViewport(deck, map, useMapboxProjection = true) {
   );
 }
 
-function afterRender(deck, map) {
-  const {mapboxLayers, isExternal} = deck.props.userData;
+function afterRender(deck: Deck, map: Map): void {
+  const {mapboxLayers, isExternal} = deck.userData as UserData;
 
   if (isExternal) {
     // Draw non-Mapbox layers
     const mapboxLayerIds = Array.from(mapboxLayers, layer => layer.id);
-    const hasNonMapboxLayers = deck.props.layers.some(layer => !mapboxLayerIds.includes(layer.id));
+    const hasNonMapboxLayers = deck.props.layers.some(
+      layer =>
+        layer &&
+        // @ts-ignore (2339) id does not exist on Layer[]
+        !mapboxLayerIds.includes(layer.id)
```

Comment:
Will this not fail if `layers` has nested layers? Instead of the ts-ignore could we flatten the layers?